### DISCUSSION TITLE
why borther define orb-preformat-keywords insdeat of using all possible kewords?

### DISCUSSION THREAD

**Discussion by wztdream** [org-roam/org-roam-bibtex#248] [Q&A]
Hi,
I am curious about the purpose of  `orb-preformat-keywords`, why not use all possible bib entry keywords?

**Comment by myshevchuk** [comment]
Hi @wztdream,

Because placeholders used for expanding templates with BibTeX keywords are in fact normal Org-capture wildcards `%^{...}`, which in Org-capture are used as prompts, it seemed reasonable to me to let the users have more of a fine control over which placeholders they wish to expand with BibTeX keywords and which they would like to use as prompts. For example, in a situation, when a user has a `%^{note}` placeholder in the templates, which they want to use as a Org-capture prompt, the placeholder should not be expanded unconditionally with a BibTeX value. _ORB does not make any assumptions on how the user is going to use Org-capture template placeholders._ It does only what it is explicitly told to do. This is a sort of the old Emacs philosophy as I understand it.

All that said, I agree that the above scenario is less than likely for a typical use case. A possible solution would be to add a special value `all` to `orb-preformat-keywords`, which would mean all the encountered placeholders will be expanded provided there is a corresponding BibTeX field. But since no one requested this until now, it's not implemented yet. You are welcome to submit a feature request.
